docs/csharp/tutorials/nullable-reference-types.md

```diff
@@ -0,0 +1,225 @@
+1. Write out important statistics on the survey responses.
+
+## Build the survey with nullable and non-nullable types
+
+The first code you'll write creates the survey. You'll write classes to model a survey question and a survey run. Your survey has three types of questions, distinguished by the format of the answer: Yes/No answers, number answers, and text answers. Create a `public` `SurveyQuestion` class. Include the `nullable enable` pragma immediately after the `using` statements:
+
+```csharp
+#nullable enable
+namespace NullableIntroduction
+{
+    public class SurveyQuestion
+    {
+    }
+}
+```
+
+Adding the `#nullable enable` pragma means the compiler interprets every reference type variable declaration as a **non-nullable** reference type. You can see your first warning by adding properties for the question text and the type of question, as shown in the following code:
+
+```csharp
+#nullable enable
+namespace NullableIntroduction
+{
+    public enum QuestionType
+    {
+        YesNo,
+        Number,
+        Text
+    }
+
+    public class SurveyQuestion
+    {
+        public string QuestionText { get; }
+        public QuestionType TypeOfQuestion { get; }
+    }
+}
+```
+
+Because you haven't initialized `QuestionText`, the compiler issues a warning that a non-nullable property hasn't been initialized. Your design requires the question text to be non-null, so you add a constructor to initialize it and the `QuestionType` value as well. The finished class definition looks like the following code:
+
+[!code-csharp[DefineQuestion](../../../samples/csharp/NullableIntroduction/NullableIntroduction/SurveyQuestion.cs)]
+
+Adding the constructor removes the warning. The constructor argument is also a non-nullable reference type, so the compiler doesn't issue any warnings.
+
+Next, create a `public` class named `SurveyRun`. Include the `nullable enable` pragma following the `using` statements. This class contains a list of `SurveyQuestion` objects and methods to add questions to the survey, as shown in the following code:
+
+```csharp
+using System.Collections.Generic;
+
+#nullable enable
+namespace NullableIntroduction
+{
+    public class SurveyRun
+    {
+        private List<SurveyQuestion> surveyQuestions = new List<SurveyQuestion>();
+
+        public void AddQuestion(QuestionType type, string question) =>
+            AddQuestion(new SurveyQuestion(type, question));
+        public void AddQuestion(SurveyQuestion surveyQuestion) => surveyQuestions.Add(surveyQuestion);
+    }
+}
+```
+
+As before, you must initialize the list object to a non-null value or the compiler issues a warning. There's no null checks in the second overload of `AddQuestion` because it's not needed: You've declared that variable to be non-nullable. Its value can't be `null`.
+
+Switch to `program.cs` in your editor, and replace the contents of `Main` with the following lines of code:
```

Comment:
isn't usually Program.cs (uppercase P) the default created file from the template?